packages/adena-extension/src/hooks/web/use-account-import-screen.ts

```diff
@@ -101,61 +146,154 @@ const useAccountImportScreen = ({ wallet }: { wallet: Wallet }): UseAccountImpor
       return null;
     }
     return { account, keyring };
-  }, [wallet, privateKey]);
+  }, [wallet, inputValue]);
 
   const onClickGoBack = useCallback(() => {
-    if (step === 'INIT') {
-      navigate(RoutePath.WebAdvancedOption);
-    } else if (step === 'SET_PRIVATE_KEY') {
-      setStep('INIT');
+    switch (step) {
+      case 'INIT':
+        navigate(RoutePath.WebAdvancedOption);
+        break;
+      case 'SET_MNEMONIC':
+        setStep('INIT');
+        break;
+      case 'SELECT_ACCOUNT':
+        setStep('SET_MNEMONIC');
+        break;
+      default:
+        break;
     }
   }, [step]);
 
   const onClickNext = useCallback(async () => {
     if (step === 'INIT') {
       if (ableToSkipQuestionnaire) {
-        setStep('SET_PRIVATE_KEY');
+        setStep('SET_MNEMONIC');
       } else {
         navigate(RoutePath.WebQuestionnaire, {
           state: {
             callbackPath: RoutePath.WebAccountImport,
           },
         });
       }
-    } else if (step === 'SET_PRIVATE_KEY') {
-      const result = await makePrivateKeyAccountAndKeyring();
-      if (!result) {
+    } else if (step === 'SET_MNEMONIC') {
```

Comment:
I wish I had confidence in life like @jinoosss has in typing out these constants correctly every time 😄

Reply:
Oh, Thanks for the good point!
I was using the string literal type.
https://github.com/onbloc/adena-wallet/pull/582/files/c3d144247b0a3feeb8c0dac741bc6bc36183ec2c#diff-fd678e1a97fbf36bebf013ce3147d3f291a7ebabed8523e436e64c90c102c021R60

I should have used the enum type, which is much clearer.